src/webhook/pkg/mutating_webhook/mutating-webhook_test.go

```diff
@@ -566,3 +566,226 @@ func TestMutatePod_NoChangeForDifferentRuntimeClass(t *testing.T) {
 		}
 	}
 }
+
+// Add test case with fractional CPU requests (eg 200m, 0.5) and limits (eg 500m, 0.5)
+func TestMutatePod_FractionalCpu(t *testing.T) {
+	// Mock environment variable
+	os.Setenv("TARGET_RUNTIMECLASS", "kata-remote")
+	os.Setenv("POD_VM_EXTENDED_RESOURCE", "kata.peerpods.io/vm")
+
+	// Create a sample pod spec with fractional CPU requests and limits
+	runtimeClassName := "kata-remote"
+	pod := &corev1.Pod{
+		Spec: corev1.PodSpec{
+			RuntimeClassName: &runtimeClassName,
+			Containers: []corev1.Container{
+				{
+					Name:  "container1",
+					Image: "busybox",
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("200m"),
+							corev1.ResourceMemory: resource.MustParse("4Gi"),
+						},
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("500m"),
+							corev1.ResourceMemory: resource.MustParse("4Gi"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	podMutator := &PodMutator{}
+	mutatedPod, err := podMutator.mutatePod(pod)
+	if err != nil {
+		t.Fatalf("mutatePod() error = %v", err)
+	}
+
+	// Check annotations
+	if mutatedPod.Annotations[PEERPODS_CPU_ANNOTATION] != "1" {
+		t.Errorf("Expected CPU annotation to be 1, got %s", mutatedPod.Annotations[PEERPODS_CPU_ANNOTATION])
+	}
+
+	if mutatedPod.Annotations[PEERPODS_MEMORY_ANNOTATION] != "4096" {
+		t.Errorf("Expected Memory annotation to be 4096, got %s", mutatedPod.Annotations[PEERPODS_MEMORY_ANNOTATION])
+	}
+
+	if _, exists := mutatedPod.Annotations[PEERPODS_GPU_ANNOTATION]; exists {
+		t.Errorf("Expected no GPU annotation, got %s", mutatedPod.Annotations[PEERPODS_GPU_ANNOTATION])
+	}
+
+	// Check resource requirements for all containers are cleared
+	// Except for the first container which has the peer-pod resource added
+	for idx, container := range mutatedPod.Spec.Containers {
```

Comment:
I just kept the loop same for all the tests. Since only the first container is expected to have the resource req/limit after mutation it will continue and exit the loop. Either we can check for number of containers as you mentioned or just use the one like in the current code.